pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -511,6 +511,10 @@ func KubeOptionObserveCreateUpdate(obj *xkube.Object) {
 	obj.Spec.ManagementPolicies = append(obj.Spec.ManagementPolicies, xpv1.ManagementActionCreate, xpv1.ManagementActionUpdate, xpv1.ManagementActionObserve)
 }
 
+func KubeOptionObserve(obj *xkube.Object) {
+	obj.Spec.ManagementPolicies = append(obj.Spec.ManagementPolicies, xpv1.ManagementActionObserve)
```

Comment:
You need to make sure the secret is only observed. Here you will probably have create and update as well.

Reply:
I checked that:
```
  managementPolicies:
  - Observe
  ```
  
  this is result in kindev, but I agree that implementation should be slightly different, I changed that. Result is the same - there's just one mgmt policy in place

Reply:
We should probably do it like this:

```suggestion
        obj.Spec.ManagementPolicies = nil
	obj.Spec.ManagementPolicies = append(obj.Spec.ManagementPolicies, xpv1.ManagementActionObserve)
```

Otherwise we'll just keep piling management policies into the array and might end up with an invalid one. See for reference: https://github.com/vshn/appcat/blob/nextcloud_collabora/pkg/comp-functions/runtime/function_mgr.go#L657